core/src/main/java/org/hibernate/ogm/loader/impl/OgmLoader.java

```diff
@@ -333,6 +360,38 @@ else if ( ogmLoadingContext.hasResultSet() ) {
 		//TODO should we collect List<Object> as result? Not necessary today
 		Object result = null;
 		List<Object> results = new ArrayList<Object>();
+
+		if ( isCollectionLoader ) {
+			// Logic to eliminate the n+1 issue in collection loading when batc fetching is enabled.
```

Comment:
Ok, looks a bit tough :) There is no more explicit way to add the referenced elements to the batch fetching queue?